app/controllers/api/base_controller/renderer.rb

```diff
@@ -266,11 +266,37 @@ def fetch_direct_virtual_attribute(type, resource, attr)
         return unless attr_accessible?(resource, attr)
         virtattr_accessor = virtual_attribute_accessor(type, attr)
         value = virtattr_accessor ? send(virtattr_accessor, resource) : virtual_attribute_search(resource, attr)
+        value = add_custom_action_hrefs(value) if attr == "custom_actions"
         result = {attr => normalize_attr(attr, value)}
         # set nil vtype above to "#{type}/#{resource.id}/#{attr}" to support id normalization
         [value, result]
       end
 
+      #
+      # WARNING: This is a hack.
```

Comment:
Prefer `HACK: ` as a lead-in to the comment.  `HACK: `  (same with `TODO: ` and `FIXME: `) are recognized by a lot of IDEs and tools to allow you to keep a checklist of things to go back to.

Reply:
I love that `# HACK:` is the official way to do it.